src/commcare_cloud/cli_utils.py

```diff
@@ -33,6 +34,16 @@ def ask_option(message, options, acceptable_responses=None):
         r = input('Please enter one of {} :'.format(', '.join(options)))
     return r
 
+def ask_string(message, args):
+    environment = get_environment(args.env_name)
+    if message not in environment.users_config.dev_users.present:
+        allowed_users = environment.users_config.dev_users.present
+        puts(color_error(
+            "Unauthorized user {}.\n\n"
+            "Please pass in one of the allowed ssh users:{}"
+            .format(message, '\n  - '.join([''] + allowed_users))))
+        return False
+    return True
```

Comment:
This should be named less generically. Also, rather than passing in `args`, should pass in `env_name` since this context does not need `args` for anything else, and indeed does not even know what it is. Finally, it should return a username rather than a boolean value.
```suggestion
def get_dev_username(env_name):
    environment = get_environment(env_name)
    username = default_username = get_default_username()
    while True:
        if not username or default_username.is_guess:
            username = input(f"Enter your SSH username ({default_username}): ")
            if not username:
                username = default_username
        if username in environment.users_config.dev_users.present:
            break
        allowed_users = environment.users_config.dev_users.present
        env_users = '\n  - '.join([''] + allowed_users)
        puts(color_error(
            f"Unauthorized user {username}.\n\n"
            f"Please pass in one of the allowed ssh users:{env_users}"
        ))
        username = ""
    if default_username.is_guess:
        print_help_message_about_the_commcare_cloud_default_username_env_var(username)
    return username
```

Reply:
Got it :-)